pkg/client/client.go

```diff
@@ -974,9 +974,20 @@ func (c *Client) WaitForDaemonSetRollout(ctx context.Context, ds *appsv1.DaemonS
 				d.Status.DesiredNumberScheduled, d.Status.UpdatedNumberScheduled)
 			return false, nil
 		}
-		if d.Status.NumberUnavailable != 0 {
-			lastErr = errors.Errorf("got %d unavailable nodes",
-				d.Status.NumberUnavailable)
+
+		var nodeReadyCount int32
+		nodeList, err := c.kclient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
```

Comment:
let's pass the context `ctx` here instead of `TODO()`

Reply:
Done